### DISCUSSION TITLE
Class "Faker\Factory" not found

### DISCUSSION THREAD

**Discussion by CushtyDev** [laravel/framework#52517] [Q&A]
### Laravel Version

11

### PHP Version

8.3

### Description

Hi I have double-checked my Laravel app with filament and there are only vendor files using the faker library I am not using it in my code, but Forge is showing this error: 

In DatabaseServiceProvider.php line 93:
Class "Faker\Factory" not found</code> 

I have seen some people say move faker to required in composer but I find it strange as I haven't used faker in my code, so do not need it in production, thanks

### Steps To Reproduce

Seed data in Forge, keep faker in dev on composer file.

**Comment by JamesFreeman** [comment]
Faker is a dev dependency - if you've deployed to forge, it won't have installed those. If you need to generate random seed data in production, you'll need to move it into a normal dependency - or don't use random data/faker in the seeders.

**Reply by CushtyDev** [reply]
Hi i am using a seeder but not using Faker, I designed my own seeders. Tha's why I am not sure why faker needs to be in required dependency. Thanks

**Reply by JamesFreeman** [reply]
Do you still have the default user seeder?

https://github.com/laravel/laravel/blob/11.x/database/seeders/DatabaseSeeder.php

**Reply by CushtyDev** [reply]
Yes I call my seeders from the default seeder, and I have a couple of user seeders there to, thanks

**Reply by JamesFreeman** [reply]
The default User seed will use a factory hence the `Faker\Factory` being called. If you remove the `::factory()` call, Does that resolve your issue?

**Comment by dofan28** [comment]
This case is like I experienced, the solution I found is not to use User::factory(1)->create(); but use User::create([ .... ]);

**Comment by ripers** [comment]
Thank you, this helped me.

I also found out that it will work if you add "fakerphp/faker": "^1.23" to the "require" section of Composer.json. 

Either that or you can run **composer install** without the **--no-dev** flag.
